src/nvim/os/fsnotification.c

```diff
@@ -0,0 +1,248 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include <assert.h>
+
+#include <uv.h>
+
+#include "nvim/vim.h"
+#include "nvim/buffer.h"
+#include "nvim/lib/khash.h"
+#include "nvim/lib/klist.h"
+#include "nvim/fileio.h"
+#include "nvim/os/fs_defs.h"
+#include "nvim/os/fsnotification.h"
+#include "nvim/os/event.h"
+
+
+// Apparentely, the destruction mechanism is not very well implemented in klist.
+// Proper resource handling will be done manually
+#define _destroy(x)
```

Comment:
I would actually suggest to move this to `klist.h`. I saw only a single other instance of `KLIST_INIT` in `loop.h` and it also uses `_noop`, so it is better to make both use `KLIST_INIT_NOFREE` to be placed in `klist.h`.